### DISCUSSION TITLE
files.find() method usage

### DISCUSSION THREAD

**Discussion by CooperGerman** [cloud-py-api/nc_py_api#86] [Q&A] [answered]
Hi,
I'm currently trying to implement some of the features of this package and i am facing some issues regarding the files.find() method.
I can't seem to properly use it and while debugging i found some code which i'm not very sure about in file `nc_py_api/files/files.py` lines 46-53.

```python
SEARCH_PROPERTIES_MAP = {
    "name:": "d:displayname",  # like, eq
    "mime": "d:getcontenttype",  # like, eq
    "last_modified": "d:getlastmodified",  # gt, eq, lt
    "size": "oc:size",  # gt, gte, eq, lt
    "favorite": "oc:favorite",  # eq
    "fileid": "oc:fileid",  # eq
}
```

For entry "name" should there be the colon inside the string ?

Thanks in advance for any help, i would really appreciate an example of this method allowing to verify if a file exists in a folder.

**Comment by bigcat88** [comment, marked as answer]
Thank for pointing on this bug.
Fix & test will arrive today for this.

**Reply by Dymirt** [reply]
Hi @bigcat88 Is there any possible way to collaborate with you on this library?

**Reply by bigcat88** [reply]
Of course there is, it's an open source project.
The easiest way that I see: the contributor chooses what he wants to do, together we determine how it will be done and after that we do a review.
If there are any difficulties, I will try to help.

**Reply by CooperGerman** [reply]
@bigcat88 Thx for the fix, i'd be fine with being credited.
